lgc/patch/PatchEntryPointMutate.cpp

```diff
@@ -110,14 +110,22 @@ void PatchEntryPointMutate::processShader() {
 
   Function *origEntryPoint = m_entryPoint;
 
+  // Create new function, empty for now.
   Function *entryPoint = Function::Create(entryPointTy, GlobalValue::ExternalLinkage, "", m_module);
   entryPoint->setCallingConv(origEntryPoint->getCallingConv());
-  entryPoint->addFnAttr(Attribute::NoUnwind);
   entryPoint->takeName(origEntryPoint);
 
-  ValueToValueMapTy valueMap;
-  SmallVector<ReturnInst *, 8> retInsts;
-  CloneFunctionInto(entryPoint, origEntryPoint, valueMap, false, retInsts);
+  // Transfer code from old function to new function.
```

Comment:
This is a good change. I have one question: how to copy blocks into multiple functions. Does it mean we couldn't simply just set new parents for such blocks?

Reply:
If you want to copy the same block into N different places, then you will need at least N-1 clones. Is that what you mean?

Reply:
Yes, this is what I mean. If we just move the blocks from one parent to another, it is cheap. If you want N copies of those blocks for N parents, cloning is still needed. I just want to confirm this.